src/wp-includes/block-template-utils.php

```diff
@@ -837,3 +872,55 @@ function block_header_area() {
 function block_footer_area() {
 	block_template_part( 'footer' );
 }
+
+/**
+ * Creates an export of the current templates and
+ * template parts from the site editor at the
+ * specified path in a ZIP file.
+ *
+ * @since 5.9.0
+ *
+ * @return WP_Error|string Path of the ZIP file or error on failure.
+ */
+function wp_generate_block_templates_export_file() {
+	if ( ! class_exists( 'ZipArchive' ) ) {
+		return new WP_Error( __( 'Zip Export not supported.', 'gutenberg' ) );
+	}
+
+	$obscura  = wp_generate_password( 12, false, false );
+	$filename = get_temp_dir() . 'edit-site-export-' . $obscura . '.zip';
+
+	$zip = new ZipArchive();
+	if ( true !== $zip->open( $filename, ZipArchive::CREATE ) ) {
+		return new WP_Error( __( 'Unable to open export file (archive) for writing.', 'gutenberg' ) );
+	}
+
+	$zip->addEmptyDir( 'theme' );
+	$zip->addEmptyDir( 'theme/block-templates' );
+	$zip->addEmptyDir( 'theme/block-template-parts' );
```

Comment:
I think the folders have been renamed now. Should we update them?

Reply:
Renamed folders.